Seperate codemod

williamlark opened the pull request:
The braid-design-system/codemod package is bundled in with braid at the moment. This script contributes 14 MB of the 16.5 MB size of braid and is only used as a helper for migrating Braid version. Splitting this package out will reduce the download size of braid-design-system and allow us to iterate on the package independently. 

~I've marked this as a **major release** since it removes functionality.~

felixhabib commented:
Probably don't wanna trigger a major release for the braid package specifically if it can be avoided? The codemod package is fine

askoufis commented on packages/codemod/package.json:
```diff
@@ -10,6 +9,12 @@
+  ],
   "author": "SEEK",
   "license": "MIT",
   "dependencies": {
```
These should all be moved to `devDependencies`. This didn't matter before since the package was private, but now that it's not we don't want consumers to install all these deps since they're already bundled with the CLI.

williamlark replied:
Good catch. Fixed up

askoufis commented:
> If it needs to be major, that's ok. But we should probably discuss when to release it. I'd prefer that current work isn't held behind a major release for our customers. Probably the sku group and others can give better feedback on if we can get away with minor

IMO this is one of those "technically major" but "practically minor" releases. The codemod is barely public API anyway, it's only "documented" inside the changelog, and when the time comes for consumers to actually run the codemod, we'll refer to the new package and it'll be fine.

Additionally, there haven't been any codemods since v31.11, so as long as a consumer doesn't skip straight to the latest version when coming from 30 or 30.10 or something, then they should be fine.

Ultimately I _think_ there's very little risk with this change.

williamlark commented:
> > If it needs to be major, that's ok. But we should probably discuss when to release it. I'd prefer that current work isn't held behind a major release for our customers. Probably the sku group and others can give better feedback on if we can get away with minor
> 
> IMO this is one of those "technically major" but "practically minor" releases. The codemod is barely public API anyway, it's only "documented" inside the changelog, and when the time comes for consumers to actually run the codemod, we'll refer to the new package and it'll be fine.
> 
> Additionally, there haven't been any codemods since v31.11, so as long as a consumer doesn't skip straight to the latest version when coming from 30 or 30.10 or something, then they should be fine.
> 
> Ultimately I _think_ there's very little risk with this change.

Agree. Sounds like were happy to do a minor for this one. I've updated the changeset with a note mentioning the minor bump and changed the changeset release type for braid.

michaeltaranto commented on packages/codemod/README.md:
```diff
@@ -0,0 +1,11 @@
+# @braid-design-system/codemod
+
+Automates migrations Braid migrations.
```
```suggestion
Automates migrations between versions of Braid Design System.
```

michaeltaranto commented on packages/codemod/README.md:
````diff
@@ -0,0 +1,11 @@
+You must provide a target Braid version and a glob to your projects source files. For example:
+
+```bash
+pnpm dlx @braid-design-system/codemod v31.11 "**/*.{ts,tsx}
````
```suggestion
pnpm dlx @braid-design-system/codemod v31.11 "**/*.{ts,tsx}"
```

michaeltaranto commented on packages/codemod/package.json:
```diff
@@ -10,16 +9,25 @@
     "format:eslint": "eslint --cache --fix .",
     "format:prettier": "prettier --cache --write ."
   },
+  "bin": {
+    "braid-upgrade": "dist/index.js"
+  },
```
Now that we are removing the `bin` file from the other directory, how are we ensuring the built file has a `#!` env is set to `node`? i.e. `#!/usr/bin/env node`

williamlark replied:
The index file includes it explicitly and it's included when built with esbuild. As long as we have it in the index.ts it will be added to the top of the output file. I tested it out to double check this was the case.
